addons/sale_stock/models/replenishment_warhouse_select.py

```diff
@@ -0,0 +1,54 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+from odoo.osv import expression
+
+
+class SaleStockReplenishmentWarehouseSelect(models.TransientModel):
+    _name = 'sale_stock.replenishment.warehouse.select'
+    _description = 'Replenishment Warehouse Select'
+    _rec_name = 'sale_order_line_id'
+
+    sale_order_line_id = fields.Many2one('sale.order.line')
+    product_id = fields.Many2one('product.product', related='sale_order_line_id.product_id')
+    qty_to_order = fields.Float(related='sale_order_line_id.product_uom_qty')
+
+    route_ids = fields.One2many('stock.route', compute='_compute_route_ids')
+    wh_replenishment_option_ids = fields.One2many('stock.replenishment.option', compute='_compute_wh_replenishment_options')
+
+    def _compute_route_ids(self):
+        customer_location = self.sale_order_line_id.order_id.partner_shipping_id.property_stock_customer
+        routes = self.env['stock.rule'].search(expression.AND([
+            self.env['stock.rule']._check_company_domain(self.env.company),
+            [('location_dest_id', '=', customer_location.id)],
+        ])).route_id
+        for warehouse_select in self:
```

Comment:
It's probably there than you can filter out the dropship. The rule should have a warehouse to be selected